### PR TITLE
[Mem control] Move one of the insert check out from the sg lock

### PR DESCRIPTION
**Pull request by HTHou**

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/engine/storagegroup/StorageGroupProcessor.java [hunk 1]
```diff
@@ -710,10 +717,17 @@ public void insert(InsertRowPlan insertRowPlan) throws WriteProcessException {
    */
   @SuppressWarnings("squid:S3776") // Suppress high Cognitive Complexity warning
   public void insertTablet(InsertTabletPlan insertTabletPlan) throws BatchInsertionException {
+    TSStatus[] results = new TSStatus[insertTabletPlan.getRowCount()];
```
Comment (mychaow) [inline server/src/main/java/org/apache/iotdb/db/engine/storagegroup/StorageGroupProcessor.java:720]: I think it's better to move the two line after blockInsertionIfReject  for that we need not to create them before we get memory.

Comment (HTHou) [reply server/src/main/java/org/apache/iotdb/db/engine/storagegroup/StorageGroupProcessor.java:720]: Sure~

**Review by mychaow — APPROVED** [review]

File: server/src/main/java/org/apache/iotdb/db/engine/storagegroup/StorageGroupProcessor.java [hunk 2]
```diff
@@ -710,6 +717,15 @@ public void insert(InsertRowPlan insertRowPlan) throws WriteProcessException {
    */
   @SuppressWarnings("squid:S3776") // Suppress high Cognitive Complexity warning
   public void insertTablet(InsertTabletPlan insertTabletPlan) throws BatchInsertionException {
+    if (enableMemControl) {
+      try {
+        blockInsertionIfReject();
+      } catch (WriteProcessException e) {
+        TSStatus[] results = new TSStatus[insertTabletPlan.getRowCount()];
+        Arrays.fill(results, RpcUtils.SUCCESS_STATUS);
```
Comment (jixuan1989) [inline server/src/main/java/org/apache/iotdb/db/engine/storagegroup/StorageGroupProcessor.java:725]: SUCCESS_STATUS？

Comment (HTHou) [reply server/src/main/java/org/apache/iotdb/db/engine/storagegroup/StorageGroupProcessor.java:725]: That is what I am confused...

Comment (HTHou) [reply server/src/main/java/org/apache/iotdb/db/engine/storagegroup/StorageGroupProcessor.java:725]: That is a mistake, I'll fix it. Thanks.

**Review by qiaojialin — APPROVED** [review]

**Pull request merged**